### DISCUSSION TITLE
Trying to reply manually for some users reply

### DISCUSSION THREAD

**Discussion by hkpmatt** [python-telegram-bot/python-telegram-bot#3843] [Q&A] [answered]
![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/70888649/cba775e5-9ab6-49d6-b472-07f286d89bcc)
My solution here is to forward the private message to a group with me.
However, i can only get the users chat id and send to the group like below
and use a syntax "/reply <chat_id> <message>"  at the telegram group to forward my manual message to the user
<img width="186" alt="image" src="https://github.com/python-telegram-bot/python-telegram-bot/assets/70888649/40cd3055-de74-44f6-9998-099c3029ec39">
Is there get other way to store the chat_id globally without replaced by new user or other function that i can just reply to the forwarded message to reply the user on the telegram group?

**Comment by hkpmatt** [comment]
<img width="455" alt="image" src="https://github.com/python-telegram-bot/python-telegram-bot/assets/70888649/a0cc9746-4350-4120-9e35-a1dd0131212a">
This is the support group of the bot.

![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/70888649/41fe686b-6940-4a8f-aa53-c91b0af92faa)

And this is what the user will receive when talking to the bot

**Comment by hkpmatt** [comment]
![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/70888649/d0e745f7-c4d0-4834-ac5a-93b42530d3bc)
is there any way that i can get the id from this Forward_From.

**Comment by harshil21** [comment, marked as answer]
You can for e.g. attach the user's id to the support message, and then while you reply to it, retrieve the replied message using `update.effective_message.reply_to_message` and parse that to get the original user's id. This way you don't need to store anything. e.g:

"""
`From (<username|name> - <user_id>): Hi i have this problem with the bot...` 

```
> From (<username|name> - <user_id>): Hi i have this problem with the bot...

Hi do this...
```
``` python
org_msg = update.effective_message.reply_to_message
user_id = parsing code here
await context.bot.send_message(user_id, update.effective_message.text)
```

**Reply by hkpmatt** [reply]
tried this code right here, it can parse to get the chat id under forward_from.
![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/70888649/c4f3882a-22bc-42f6-a44a-d6ee4ffeaf11)

now I can directly reply to the user via the bot when user dont hide their forward message
```
code = str(update.effective_message.reply_to_message)
forward_from_start = code.find("forward_from=User(") + len("forward_from=User(")
forward_from_end = code.find(", from_user=")

# Extract the substring containing forward_from field
forward_from_str = code[forward_from_start:forward_from_end]

# Find the start and end indices of the id field within forward_from
id_start = forward_from_str.find("id=") + len("id=")
id_end = forward_from_str.find(",", id_start)

# Extract the id value
id_value = forward_from_str[id_start:id_end]
print(id_value)
```

**Reply by harshil21** [reply]
well my point was to not forward the user's message, but to just extract it and send it to you along with user id (`update.effective_user.id`) (`update.effective_message.text`). This way you bypass the forward secrecy issue.

**Reply by hkpmatt** [reply]
Yes this is my original solution.
The number after '/reply' is the id from the sender.
But I want to directly reply to the forwarded message to reply the user
